### PR TITLE
feat(schematics): add node-app --framework nestjs

### LINKED ISSUE #105: Improve compatibility with Nx
## I'm submitting a...
<pre><code>[ ] Regression
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request</code></pre>

## Current behavior
NestJS schematics are currently incompatible with Nx.

## Expected behavior
As per the [pull request](https://github.com/nrwl/nx/pull/807) by @FallenRiteMonk over at the Nx repository, NestJS should be compatible with Nx.

## What is the motivation / use case for changing the behavior?
@FrozenPandaz mentioned that they have reached out to @kamilmysliwiec regarding the above mentioned pull request, and that he has expressed interest in making this happen. Whether that comes to fruition is to be decided, but I was hoping to at least start a conversation, as both Nx and NestJS are invaluable assets to many of us.

## Environment
<pre><code>Nest version: 5.4.0</code></pre>

### PR DESCRIPTION
**Pull request by FrozenPandaz**
## Current Behavior

NestJS is a great choice for enterprises using Angular who are also looking to build backends.

Creating Nest Applications with the nest schematics [has some friction](https://github.com/nestjs/schematics/issues/105#issuecomment-446655741)

## Expected Behavior

A NestJS Application can be created via

```sh
ng g node-app nest-app --framework nestjs
ng g @nestjs/schematics:controller cats --source-root apps/nest-app/src --path app
```

## Notes

This is based off of the work done by @FallenRiteMonk in https://github.com/nrwl/nx/pull/807

@FallenRiteMonk 
Thank you so much for your work. Sorry it was not accepted as is at the time. 

## Issues

Fixes https://github.com/nestjs/schematics/issues/105

### CODE REVIEW

**Comment by FrozenPandaz** [comment]
cc @kamilmysliwiec

**Review by vsavkin — CHANGES_REQUESTED** [review]
Looks good. A few minor comments.

File: packages/schematics/src/collection/node-application/files/nestjs/main.ts__tmpl__ [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+  await app.listen(3333);
+}
+
+(async () => {
+  await bootstrap();
+  console.log('Listening at http://localhost:3333');
+})();
```
Comment (kamilmysliwiec) [inline packages/schematics/src/collection/node-application/files/nestjs/main.ts__tmpl__]: Why not
```typescript
import { NestFactory } from '@nestjs/core';

import { AppModule } from './app.module';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  await app.listen(3000, () => console.log('Listening at http://localhost:3333'));
}
bootstrap();
```

**Comment by kamilmysliwiec** [comment]
@nestjs app initial structure looks good @FrozenPandaz, looking forward to see this feature in Nx :)

File: packages/schematics/src/collection/node-application/files/nestjs/main.ts__tmpl__ (hunk 1 above)
Comment (FrozenPandaz) [reply packages/schematics/src/collection/node-application/files/nestjs/main.ts__tmpl__]: Looks good, done!

**Pull request merged**